rpyc/rpyc_client/command_handlers.py

```diff
@@ -91,84 +82,108 @@ def _execute(self):
-        return remote_os.stat(path), None
+        return remote_os.stat(path)
 
 
 class KillProcessHandler(CommandHandler):
-    def _parse_input(self, split_input, additional_input):
-        if len(split_input) < 2:
+    def __init__(self, rpyc_conn, created_processes):
+        super().__init__(rpyc_conn)
+        self.created_processes = created_processes
+
+    def __parse_input(self, command_with_arguments):
+        parser = argparse.ArgumentParser(command_with_arguments[0])
+        parser.add_argument('path')
+        args = parser.parse_args(command_with_arguments[1:])
+        return args.path
+
+    def execute(self, command_with_arguments):
+        if len(command_with_arguments) < 2:
             raise TypeError('expects at least one argument')
 
-        if len(split_input) == 3:
-            if not split_input[1].startswith('-'):
-                raise ValueError("kill signal number should start with '-'")
-            return {'signal': int(split_input[1][1:]), 'pid': int(split_input[2])}
-        elif len(split_input) == 2:
-            return {'pid': int(split_input[1])}
+        if len(command_with_arguments) == 3:
+            return self.__execute_kill_signo_pid(command_with_arguments)
+
+        if len(command_with_arguments) != 2:
+            raise TypeError("Incorrect usage")
+
+        if command_with_arguments[1] == 'all':
+            return self.__execute_kill_all()
 
-        raise TypeError('too many arguments')
+        return self.__execute_kill_pid(command_with_arguments)
 
-    def _execute(self, pid, signal=9):
+    def __execute_kill_signo_pid(self, command_with_arguments):
+        if not command_with_arguments[1].startswith('-'):
+            raise ValueError('Incorrect usage')
+        signo = int(command_with_arguments[1][1:])
+        pid = int(command_with_arguments[2])
+        return self.__kill(pid, signo)
+
+    def __execute_kill_pid(self, command_with_arguments):
+        pid = int(command_with_arguments[1])
+        return self.__kill(pid)
+
+    def __kill(self, pid, signal=9):
         remote_os = self.rpyc_conn.modules.os
         remote_os.kill(pid, signal)
-        return self.empty_output
+
+    def __execute_kill_all(self):
+        for process in self.created_processes:
+            process.kill()
+        self.created_processes.clear()
 
 
 class RunAsNewProcessHandler(CommandHandler):
-    def _parse_input(self, split_input, additional_input):
-        if len(split_input) < 2:
-            raise TypeError('expecting at least one argument')
-        return {'path_to_executable': split_input[1], 'argv_to_exe': split_input[2:]}
+    def __init__(self, rpyc_conn, created_processes):
+        super().__init__(rpyc_conn)
+        self.created_processes = created_processes
 
-    def _execute(self, path_to_executable, argv_to_exe):
+    def execute(self, command_with_arguments):
+        path_to_exe, argv_to_exe = parse_single_path_argument(command_with_arguments)
```

Comment:
but parse_single_path_argument returns only the path

Reply:
Geez, I though I tested this :O